EDDiscovery/UserControls/UserControlScanGrid.cs

```diff
@@ -136,119 +136,152 @@ void DrawSystem(HistoryEntry he, bool force)
+                        else if (sn.ScanData.nSemiMajorAxis.HasValue)
                         {
-                            double? g = sn.ScanData.nSurfaceGravity / JournalScan.oneGee_m_s2;
-                            Gg = " (G: " + g.Value.ToString("N1") + ")";
+                            if (sn.ScanData.IsStar || sn.ScanData.nSemiMajorAxis.Value > JournalScan.oneAU_m / 10)
+                                bdDist.AppendFormat("{0:0.00}AU ({1:0.00}ls)", (sn.ScanData.nSemiMajorAxis.Value / JournalScan.oneAU_m), sn.ScanData.nSemiMajorAxis.Value / JournalScan.oneLS_m);
+                            else
+                                bdDist.AppendFormat("{0}km", (sn.ScanData.nSemiMajorAxis.Value / 1000).ToString("N1"));
                         }
 
-                        bdDetails.Append("Landable".Tx(this) + Gg + ". ");
-                    }
+                        // display stellar bodies mass, in sols
+                        if (sn.ScanData.nStellarMass.HasValue)
+                            bdDetails.Append("Mass".Tx(this) + ": " + sn.ScanData.nStellarMass.Value.ToString("N2") + ", ");
 
-                    // append the terraformable state to the planet class
-                    if (sn.ScanData.Terraformable == true)
-                        bdDetails.Append("Terraformable".Tx(this) + ". ");
+                        // display stellar bodies radius in sols
+                        if (sn.ScanData.nRadius.HasValue)
+                            bdDetails.Append("Radius".Tx(this) + ": " + (sn.ScanData.nRadius.Value / JournalScan.oneSolRadius_m).ToString("N2") + ", ");
 
-                    // tell us that there is some volcanic activity
-                    if (sn.ScanData.Volcanism != null)
-                        bdDetails.Append("Volcanism".Tx(this) + ". ");
+                        // show the temperature
+                        if (sn.ScanData.nSurfaceTemperature.HasValue)
+                            bdDetails.Append("Temperature".Tx(this) + ": " + (sn.ScanData.nSurfaceTemperature.Value) + "K.");
 
-                    // have some ring?
-                    if (sn.ScanData.HasRings && sn.ScanData.IsStar == false)
+                        // habitable zone for stars - do not display for black holes.
+                        if (sn.ScanData.HabitableZoneInner != null && sn.ScanData.HabitableZoneOuter != null && sn.ScanData.StarTypeID != EDStar.H)
+                            bdDetails.AppendFormat(Environment.NewLine + "Habitable Zone".Tx(this) + ": {0}-{1}AU ({2}). ", (sn.ScanData.HabitableZoneInner.Value / JournalScan.oneAU_LS).ToString("N2"), (sn.ScanData.HabitableZoneOuter.Value / JournalScan.oneAU_LS).ToString("N2"), sn.ScanData.GetHabZoneStringLs());                        
+                    }
+                    else
                     {
-                        if (sn.ScanData.Rings.Count() > 1)
+                        // is a non-stellar body
+                        if (sn.ScanData.IsStar == false)
```

Comment:
This is superfluous (in this block we already know that `sn.ScanData.IsStar` is false), and wraps only the `if` statement below

Reply:
Yes, indeed... ;)